Feature/fix responsive overrides #17

tdhooper opened the pull request:
Fixes #17

simonsmith commented:
@tdhooper Sorry for my delay here. Will review this today or tomorrow. Thanks for your help again!

simonsmith commented on lib/index.js:
```diff
@@ -159,28 +151,58 @@ function combineMediaQuery(queries, resolutions) {
+// * The background definition has at-2x applied
+function extractRetinaImage(identifier, background) {
+  const match = background.match(imageRegex);
+  const [image, type, value] = match;
```
Do we need to check if match could be `null` here?

tdhooper replied:
Good spot!

simonsmith commented:
Great stuff. Cheers

simonsmith commented on lib/index.js:
```diff
@@ -159,28 +151,63 @@ function combineMediaQuery(queries, resolutions) {
+    opened += open - close;
+  });
+  return backgrounds;
+}
```
@tdhooper Curious about the intention of this function. Could the same not be achieved with `return value.split(',')`?

tdhooper commented:
That was my initial approach, but it returns separate parts for the linear-gradient syntax.

Then again, the rest of the code should cope fine with that now.

> On 5 Jul 2017, at 10:04 pm, Simon Smith <notifications@github.com> wrote:
> 
> @simonsmith commented on this pull request.
> 
> In lib/index.js:
> 
> > +
> +function splitMultipleBackgrounds(value) {
> +  const backgrounds = [];
> +  let opened = 0;
> +  value.split(',').forEach((part) => {
> +    if (opened > 0) {
> +      backgrounds[backgrounds.length - 1] += `,${part}`;
> +    } else {
> +      backgrounds.push(part);
> +    }
> +    const open = count(part, '(');
> +    const close = count(part, ')');
> +    opened += open - close;
> +  });
> +  return backgrounds;
> +}
> @tdhooper Curious about the intention of this function. Could the same not be achieved with return value.split(',')?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

tdhooper commented:
Looking at this a bit more, for the given css:

```css
.multi {
  background: url(http://example.com/image.png),
                    linear-gradient(to right, rgba(255, 255, 255, 0),  rgba(255, 255, 255, 1)),
                    green,
                    url(/public/images/cool.png) at-2x;
}
```

I don't think we should repeat `green` in the `background-image` rule we create, as it's not a valid image.

Instead, `none` makes more sense:

```css
.multi {
  background-image: url(http://example.com/image.png),
                    linear-gradient(to right, rgba(255, 255, 255, 0),  rgba(255, 255, 255, 1)),
                    none,
                    url(/public/images/cool@2x.png);
}
```
